src/resolvers/Appointment.js

```diff
@@ -0,0 +1,102 @@
+import { ApolloError, UserInputError } from "apollo-server-express";
+import Appointment from "../models/Appointment";
+import User from "../models/User";
+import Branch from "../models/Branch";
+import Service from "../models/Service";
+
+const AppointmentStatus = {
+  "PENDING": "PENDING",
+  "REJECTED": "REJECTED",
+  "ACCEPTED": "ACCEPTED",
+  "COMPLETED": "COMPLETED",
+  "CANCELLED": "CANCELLED"
+}
+
+const AppointmentResolver = {
+  Query: {
+    appointments: async(root, { filter } , context, info) => {
+      if (!filter){
+        throw new UserInputError("No filter provided");
+      }
+      try{
+        const filteredAppointments = await Appointment.find(JSON.parse(filter));
+        return filteredAppointments;

```

Comment:
1. parse filter -> try catch
2. find by filter
do not combine 2 operations, this will caused caught error to be implicit